airflow/www/views.py

```diff
@@ -258,7 +258,11 @@ def task_group_to_grid(task_item_or_group, dag, dag_runs, session):
     if isinstance(task_item_or_group, AbstractOperator):
         return {
             'id': task_item_or_group.task_id,
-            'instances': [wwwutils.get_task_summary(dr, task_item_or_group, session) for dr in dag_runs],
+            'instances': [
+                ts
+                for ts in [wwwutils.get_task_summary(dr, task_item_or_group, session) for dr in dag_runs]
+                if ts is not None
+            ],
```

Comment:
```suggestion
            'instances': [
                ts
                for ts in (wwwutils.get_task_summary(dr, task_item_or_group, session) for dr in dag_runs)
                if ts is not None
            ],
```

This should save some memory